Sources/Controllers/2.0/Settings/OABaseSettingsWithBottomButtonsViewController.mm

```diff
@@ -0,0 +1,179 @@
+//
+//  OABaseSettingsWithBottomButtonsViewController.mm
+//  OsmAnd Maps
+//
+//  Created by Anna Bibyk on 15.10.2020.
+//  Copyright © 2020 OsmAnd. All rights reserved.
+//
+
+#import "OABaseSettingsWithBottomButtonsViewController.h"
+#import "Localization.h"
+#import "OAColors.h"
+#include <OsmAndCore/Utilities.h>
+
+#define kTopBottomPadding 8
+#define kButtonTopBottomPadding 10
+#define kButtonSidePadding 16
+#define kSidePadding 16
+#define kTopPadding 6
+#define kBottomPadding 32
+
+
+@implementation OABaseSettingsWithBottomButtonsViewController
+{
+    CGFloat _heightForHeader;
+}
+
+- (instancetype) init
+{
+    self = [super initWithNibName:@"OABaseSettingsWithBottomButtonsViewController" bundle:nil];
+    if (self)
+    {
+    }
+    return self;
+}
+
+- (void) viewDidLoad
+{
+    [super viewDidLoad];
+    [self setupNavbar];
+    [self setupBottomView];
+}
+
+- (void) setupNavbar
+{
+    BOOL hasImage = !self.backImageButton.hidden;
+    
+    self.cancelButtonLeftMarginWithIcon.active = hasImage;
+    self.cancelButtonLeftMarginNoIcon.active = !hasImage;
+}
+
+- (void) setupBottomView
+{
+    BOOL hasPrimaryButton = !self.primaryBottomButton.hidden;
+    BOOL hasSecondaryButton = !self.secondaryBottomButton.hidden;
+    BOOL hasBottomView = !self.bottomBarView.hidden;
+    
+    self.primaryBottomButton.layer.cornerRadius = 9.;
+    self.secondaryBottomButton.layer.cornerRadius = 9.;
+    
+    self.primaryButtonTopMarginYesSecondary.active = hasPrimaryButton && hasSecondaryButton;
+    self.primaryButtonTopMarginNoSecondary.active = !hasSecondaryButton;
+    
+    self.bottomViewHeigh.constant = (hasPrimaryButton ? self.primaryButtonHeight.constant : 0.) + (hasSecondaryButton ? self.secondaryButtonHeight.constant : 0.) + kTopBottomPadding * (hasPrimaryButton && hasSecondaryButton ? 3 : 2) + [OAUtilities getBottomMargin];
+    
+    self.tableViewBottomMarginYesView.active = hasBottomView;
+    self.tableViewBottomMarginNoView.active = !hasBottomView;
+}
+
+- (void) setToButton:(UIButton *)button firstLabelText:(NSString *)firstLabelText firstLabelFont:(UIFont *)firstLabelFont firstLabelColor:(UIColor *)firstLabelColor secondLabelText:(NSString *)secondLabelText secondLabelFont:(UIFont *)secondLabelFont secondLabelColor:(UIColor *)secondLabelColor
```

Comment:
Isn't it better to pass these attributes as a dictionary with style parameters (like you do in NSDictionary *firstLabelAttributes)